src/test/fuzz/validation_load_mempool.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparamsbase.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/setup_common.h>
+#include <txmempool.h>
+#include <util/time.h>
+#include <validation.h>
+
+#include <cstdint>
+#include <vector>
+
+namespace {
+FuzzedFileProvider* fuzzed_file_provider_ptr = nullptr;
+
+FILE* fuzzed_fopen(const fs::path&, const char*)
+{
+    return fuzzed_file_provider_ptr->open();
+}
+} // namespace
+
+void initialize_validation_load_mempool()
+{
+    static const auto testing_setup = MakeNoLogFileContext<const TestingSetup>();
+}
+
+FUZZ_TARGET_INIT(validation_load_mempool, initialize_validation_load_mempool)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    SetMockTime(ConsumeTime(fuzzed_data_provider));
+    FuzzedFileProvider fuzzed_file_provider = ConsumeFile(fuzzed_data_provider);
+    fuzzed_file_provider_ptr = &fuzzed_file_provider;
+
+    CTxMemPool pool{};
+    (void)LoadMempool(pool, ::ChainstateActive(), fuzzed_fopen);
```

Comment:
would be nice to not introduce a new global

Reply:
Good point! Now addressed.

Reply:
Not addressed?

nvm. @dongcarl will fix this :grimacing:

Reply:
I thought you meant the introduction of `fuzzed_fopen`? What did you mean? :)

Reply:
Oh, the ::ChainstateActive()